kernel/kernel-api/src/main/java/io/delta/kernel/internal/skipping/DataSkippingUtils.java

```diff
@@ -260,4 +311,54 @@ private static Predicate reverseComparatorFilter(Predicate predicate) {
             getLeft(predicate)
         );
     }
+
+    /**
+     * Prunes the given schema according to the provided set of top-level columns to keep. The
+     * columns in {@code schema} are taken to be nested under {@code parentPath}.
+     *
+     * @param schema schema to prune
+     * @param columnsToKeep set of top-level columns to keep
+     * @param parentPath parent path of the fields in {@code schema} relative to the top-level
+     *                   schema
+     */
+    private static StructType pruneSchema(
```

Comment:
this method `pruneSchema` seems to accomplish the goal of the PR: `Prune the statistics schema based on the columns used in the generated data skipping filter`

What are all of the `DataSkippingPredicate` changes needed for?

Reply:
The changes gather the list of referenced columns while constructing the predicate. This is how delta-spark does it; although I guess if we wanted to it could be separate step